[BSO] Colosseum QOL + Multi attempts per trip

nwjgit opened the pull request:
### Description:
Various Colosseum QOL, Custom BSO gear addition, multi attempt trips
Community polls: 
- https://discord.com/channels/342983479501389826/1032668754561224734/1268273560921575507 
- https://discord.com/channels/342983479501389826/1032668754561224734/1268286327175122986 
- https://discord.com/channels/342983479501389826/1032668754561224734/1268620754858147842
### Changes:
- Add Smol heredit to PMB
- Multi attempts per trip
- Update required gear to allow custom BSO gear & update fail message to show all possible gear substitutes
- New BSO boosts:
  - Axe of the high sungod +40% speed
  - Hellfire bow +22% speed
  - Void staff +12% speed
  - Gorajan melee gear +10% speed
  - Brawler'shook necklace +5% speed (same as torture boost)
  - Infernal bulwark -20% death reduction 
- Create a new function to handle wavetime
  - This allows for calculating max quantity for multi trips
- Send/Return messages made to look more appealing
- Update Combat Achievement code to be more specific on the chance to receive a CA
  - pass in the index number to hasChance to check each death wave in colo options
  - update the 3 relevant CA
- Update/Add a few simItems
- create a new function called gorajanGearBoost() so it can be used outside minionKill.ts
  - moved relevant code to it's own file to prevent circular dependencies
  - used to calculate gora boost for colossum
### Other checks:
- [X] I have tested all my changes thoroughly.
Went through 1 round of testing with testers and correct feedback/bugs found.
fixes for BSO: https://github.com/oldschoolgg/oldschoolbot/issues/5974
addresses part of: https://github.com/oldschoolgg/oldschoolbot/issues/6043
If this looks good/gets merged I can open a second PR to merge all the quantity/CA stuff into OSB aswell.

nwjgit commented on src/lib/colosseum.ts:
```diff
@@ -461,7 +506,7 @@ export const startColosseumRun = (options: {
 	throw new Error('Colosseum run did not end correctly.');
```
This is handled on L647 where it loops through quantity to get all the resources need and then on L758 where specialRemoveItems will handle any issues with the user not having the correct amount of supplies

gc commented:
Will merge when conflicts resolved (I just added void staff/hellfire bow) and tested

nwjgit commented:
> Will merge when conflicts resolved (I just added void staff/hellfire bow) and tested

I resolved the issues and also added the code from: https://github.com/oldschoolgg/oldschoolbot/pull/6004 to prevent any missmatch with ca

nwjgit commented:
![Discord_tNrGVEvj5R](https://github.com/user-attachments/assets/aed74fcd-fb90-4413-bfd8-7ba93845d920)
